packages/core/lib/commands/test/run.js

```diff
@@ -68,19 +68,50 @@ module.exports = async function (options) {
     inputFile: file
   });
 
-  if (config.networks[config.network]) {
+  /**
+   * If test network exists and provider, host or url doesn't exist, then internal ganache is started with the user specified
+   * configurations in the config.
+   * If development or test network with provider, host or url exists, then it connects to that network specified in the config.
+   * If no network is specified in the config, then internal ganache is started with default configuration.
+   */
+  const configuredNetwork = config.networks[config.network];
+  const testNetworkDefined = configuredNetwork && config.network == "test";
```

Comment:
I think this variable could use a better name; `testNetworkDefined` sounds like it refers to the `test` network, well, being defined, whereas what you're actually checking for is if it's defined *and* used.

Reply:
I would say that generally in the Truffle codebase prefer `===` over `==` in most cases unless you actually want the behavior of `==`.

Reply:
Sorry @eggplantzzz . I forgot about this again. I will correct it.

Reply:
ooh, yea, that's a good point on the variable name